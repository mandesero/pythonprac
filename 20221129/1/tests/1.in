class C(metaclass=dump):
    def __init__(self, val):
        self.val = val

    def add(self, other, another=None):
        return self.val + other + (another or self.val)

    def boba(self):
        print("Boba")


c = C(10)
print(c.add(9))
print(c.add(9, another=11))
c.new_f = lambda x, a=1: x + 1 + a

print(c.new_f(6, 3))

def pupa(a, b):
    return a + b

c.pupa = pupa
print(c.pupa(3, 11))

c.boba()
