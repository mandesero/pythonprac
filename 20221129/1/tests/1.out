__init__: (10,), {}
add: (9,), {}
29
add: (9,), {'another': 11}
30
<lambda>: (6, 3), {}
10
pupa: (3, 11), {}
14
boba: (), {}
Boba
